core/app/service/setting.go

```diff
@@ -394,6 +377,25 @@ func (u *SettingService) UpdatePassword(c *gin.Context, old, new string) error {
 	return nil
 }
 
+func (u *SettingService) UpdateSystemSSL() error {
+	certPath := path.Join(global.CONF.System.BaseDir, "1panel/secret/server.crt")
+	keyPath := path.Join(global.CONF.System.BaseDir, "1panel/secret/server.key")
+	certificate, err := os.ReadFile(certPath)
+	if err != nil {
+		return err
+	}
+	key, err := os.ReadFile(keyPath)
+	if err != nil {
+		return err
+	}
+	cert, err := tls.X509KeyPair(certificate, key)
+	if err != nil {
+		return err
+	}
+	constant.CertStore.Store(&cert)
+	return nil
+}
+
 func loadInfoFromCert() (dto.SSLInfo, error) {
 	var info dto.SSLInfo
 	certFile := path.Join(global.CONF.System.BaseDir, "1panel/secret/server.crt")
```

Comment:
The provided code looks mostly consistent from a design and implementation standpoint across the files, with no obvious errors or irregularities. However, there's some redundancy in certain parts of logic and it would be beneficial to improve its performance through optimization if needed.

I can make a few suggestions:

1. Reuse variable assignments wherever possible within an `if` statement.
   
2. Check file existence using `fileExists()`, which would prevent infinite loop when trying to remove non-existent paths like server.crt & key.pem.

3. Simplify certificate management functions by removing redundant imports.

4. Consider adding error handling inside individual function calls where appropriate (e.g., after successful execution).

5. Improve readability by grouping related operations into simpler methods if necessary.

With those points considered for future improvements, we do not identify any significant issues at this moment during the comparison.